### PR TITLE
US-112 | fix: fix unknown and zero accessibility shortcoming counts

### PR DESCRIPTION
**Pull request by karisal-anders**
### fix: fix unknown and zero accessibility shortcoming counts

refs US-112

## Tickets

[US-112](https://helsinkisolutionoffice.atlassian.net/browse/US-112)

## Manual testing with venue ID 5

Set up data locally:
- `docker compose up --build`
- `docker exec -it unified-search-sources-1 bash`
  - `python manage.py ingest_data location`

Opened http://localhost:4000/search in Insomnia and queried with:

<summary>GraphQL query</summary>

<details>

```graphql
query unifiedSearch {
  unifiedSearch(
    index: "location"
    q: "Kaunokki"
    languages: [FINNISH]) {
    edges {
      cursor
      node {
        venue {
          meta {
            id
          }
          name {
            fi
          }
          accessibility {
            viewpoints {
              id
              value
              name {
                fi
              }
              shortages {
                fi
              }
            }
            shortcomings {
              profile
              count
            }
          }
        }
      }
    }
  }
}
```
</details>

<summary>Results</summary>

<details>

```json
{
  "data": {
    "unifiedSearch": {
      "edges": [{
          "cursor": "eyJvZmZzZXQiOjF9",
          "node": {
            "venue": {
              "meta": {
                "id": "5"
              },
              "name": {
                "fi": "Päiväkoti Kaunokki"
              },
              "accessibility": {
                "viewpoints": [{
                    "id": "11",
                    "value": "red",
                    "name": {
                      "fi": "Olen pyörätuolin käyttäjä"
                    },
                    "shortages": [{
                        "fi": "Kulkureitti sisäänkäynnille ei ole tasainen."
                      }, {
                        "fi": "Kulkureitillä sisäänkäynnille on jyrkkä mäki."
                      }, {
                        "fi": "Sisäänkäynnissä on yli 2 cm korkea kynnys."
                      }, {
                        "fi": "Sisäänkäynnissä on ahdas tuulikaappi."
                      }
                    ]
                  }, {
                    "id": "12",
                    "value": "red",
                    "name": {
                      "fi": "Olen pyörätuolin käyttäjä - saavun omalla autolla"
                    },
                    "shortages": [{
                        "fi": "Esteettömiä autopaikkoja ei ole."
                      }, {
                        "fi": "Kulkureitti sisäänkäynnille ei ole tasainen."
                      }, {
                        "fi": "Kulkureitillä sisäänkäynnille on jyrkkä mäki."
                      }, {
                        "fi": "Sisäänkäynnissä on yli 2 cm korkea kynnys."
                      }, {
                        "fi": "Sisäänkäynnissä on ahdas tuulikaappi."
                      }
                    ]
                  }, {
                    "id": "13",
                    "value": "red",
                    "name": {
                      "fi": "Olen pyörätuolin käyttäjä - saavun saattoliikenteellä"
                    },
                    "shortages": [{
                        "fi": "Kulkureitti sisäänkäynnille ei ole tasainen."
                      }, {
                        "fi": "Kulkureitillä sisäänkäynnille on jyrkkä mäki."
                      }, {
                        "fi": "Sisäänkäynnissä on yli 2 cm korkea kynnys."
                      }, {
                        "fi": "Sisäänkäynnissä on ahdas tuulikaappi."
                      }
                    ]
                  }, {
                    "id": "21",
                    "value": "red",
                    "name": {
                      "fi": "Olen liikkumisesteinen, mutta kävelen"
                    },
                    "shortages": [{
                        "fi": "Kulkureitti sisäänkäynnille ei ole tasainen."
                      }, {
                        "fi": "Kulkureitillä sisäänkäynnille on jyrkkä mäki."
                      }
                    ]
                  }, {
                    "id": "22",
                    "value": "red",
                    "name": {
                      "fi": "Olen liikkumisesteinen, mutta kävelen - saavun omalla autolla"
                    },
                    "shortages": [{
                        "fi": "Esteettömiä autopaikkoja ei ole."
                      }, {
                        "fi": "Kulkureitti sisäänkäynnille ei ole tasainen."
                      }, {
                        "fi": "Kulkureitillä sisäänkäynnille on jyrkkä mäki."
                      }
                    ]
                  }, {
                    "id": "23",
                    "value": "red",
                    "name": {
                      "fi": "Olen liikkumisesteinen, mutta kävelen - saavun saattoliikenteellä"
                    },
                    "shortages": [{
                        "fi": "Kulkureitti sisäänkäynnille ei ole tasainen."
                      }, {
                        "fi": "Kulkureitillä sisäänkäynnille on jyrkkä mäki."
                      }
                    ]
                  }, {
                    "id": "31",
                    "value": "red",
                    "name": {
                      "fi": "Olen rollaattorin käyttäjä"
                    },
                    "shortages": [{
                        "fi": "Kulkureitti sisäänkäynnille ei ole tasainen."
                      }, {
                        "fi": "Kulkureitillä sisäänkäynnille on jyrkkä mäki."
                      }, {
                        "fi": "Sisäänkäynnissä on yli 2 cm korkea kynnys."
                      }, {
                        "fi": "Sisäänkäynnissä on ahdas tuulikaappi."
                      }
                    ]
                  }, {
                    "id": "32",
                    "value": "red",
                    "name": {
                      "fi": "Olen rollaattorin käyttäjä - saavun omalla autolla"
                    },
                    "shortages": [{
                        "fi": "Sisäänkäynnissä on yli 2 cm korkea kynnys."
                      }, {
                        "fi": "Sisäänkäynnissä on ahdas tuulikaappi."
                      }, {
                        "fi": "Esteettömiä autopaikkoja ei ole."
                      }, {
                        "fi": "Kulkureitti sisäänkäynnille ei ole tasainen."
                      }, {
                        "fi": "Kulkureitillä sisäänkäynnille on jyrkkä mäki."
                      }
                    ]
                  }, {
                    "id": "33",
                    "value": "red",
                    "name": {
                      "fi": "Olen rollaattorin käyttäjä - saavun saattoliikenteellä"
                    },
                    "shortages": [{
                        "fi": "Kulkureitti sisäänkäynnille ei ole tasainen."
                      }, {
                        "fi": "Kulkureitillä sisäänkäynnille on jyrkkä mäki."
                      }, {
                        "fi": "Sisäänkäynnissä on yli 2 cm korkea kynnys."
                      }, {
                        "fi": "Sisäänkäynnissä on ahdas tuulikaappi."
                      }
                    ]
                  }, {
                    "id": "41",
                    "value": "green",
                    "name": {
                      "fi": "Minulla on lastenvaunut"
                    },
                    "shortages": [{
                        "fi": "Ei puutteita."
                      }
                    ]
                  }, {
                    "id": "51",
                    "value": "red",
                    "name": {
                      "fi": "Olen näkövammainen"
                    },
                    "shortages": [{
                        "fi": "Sisätilojen kulkureiteillä ei ole liikkumista ohjaavaa pintamateriaalia (tummuus- tai tuntokontrasti)."
                      }, {
                        "fi": "Sisätilojen lasiovissa ei ole kontrastimerkintöjä."
                      }
                    ]
                  }, {
                    "id": "52",
                    "value": "red",
                    "name": {
                      "fi": "Olen näkövammainen - saavun saattoliikenteellä"
                    },
                    "shortages": [{
                        "fi": "Sisätilojen lasiovissa ei ole kontrastimerkintöjä."
                      }, {
                        "fi": "Sisätilojen kulkureiteillä ei ole liikkumista ohjaavaa pintamateriaalia (tummuus- tai tuntokontrasti)."
                      }
                    ]
                  }
                ],
                "shortcomings": [{
                    "profile": "hearing_aid",
                    "count": null
                  }, {
                    "profile": "reduced_mobility",
                    "count": 3
                  }, {
                    "profile": "rollator",
                    "count": 5
                  }, {
                    "profile": "stroller",
                    "count": 0
                  }, {
                    "profile": "visually_impaired",
                    "count": 2
                  }, {
                    "profile": "wheelchair",
                    "count": 5
                  }
                ]
              }
            }
          }
        }
      ]
    }
  }
}
```
</details>

#### TPREK Accessibility viewpoints
https://www.hel.fi/palvelukarttaws/rest/v4/accessibility_viewpoint/

<summary>Results</summary>
<details>

```json
[{
        "id": "00",
        "name_fi": "Valitse esteettömyysnäkökulma",
        "name_sv": "Välj tillgänglighetsperspektiv",
        "name_en": "Choose accessibility perspective",
        "values": ["unknown"]
    }, {
        "id": "11",
        "name_fi": "Olen pyörätuolin käyttäjä",
        "name_sv": "Jag är en rullstolsanvändare",
        "name_en": "I am a wheelchair user",
        "values": ["unknown", "green", "red"]
    }, {
        "id": "12",
        "name_fi": "Olen pyörätuolin käyttäjä - saavun omalla autolla",
        "name_sv": "Jag är en rullstolsanvändare - kommer med min egen bil",
        "name_en": "I am a wheelchair user - arrive by my car",
        "values": ["unknown", "green", "red"]
    }, {
        "id": "13",
        "name_fi": "Olen pyörätuolin käyttäjä - saavun saattoliikenteellä",
        "name_sv": "Jag är en rullstolsanvändare - kommer med angöringstrafik",
        "name_en": "I am a wheelchair user - arrive with pick-up and drop-off traffic",
        "values": ["unknown", "green", "red"]
    }, {
        "id": "21",
        "name_fi": "Olen liikkumisesteinen, mutta kävelen",
        "name_sv": "Jag är rörelsehindrad, men jag går",
        "name_en": "I have reduced mobility, but I walk",
        "values": ["unknown", "green", "red"]
    }, {
        "id": "22",
        "name_fi": "Olen liikkumisesteinen, mutta kävelen - saavun omalla autolla",
        "name_sv": "Jag är rörelsehindrad, men jag går - kommer med min egen bil",
        "name_en": "I have reduced mobility, but I walk - arrive by my car",
        "values": ["unknown", "green", "red"]
    }, {
        "id": "23",
        "name_fi": "Olen liikkumisesteinen, mutta kävelen - saavun saattoliikenteellä",
        "name_sv": "Jag är rörelsehindrad, men jag går - kommer med angöringstrafik",
        "name_en": "I have reduced mobility, but I walk - arrive with pick-up and drop-off traffic",
        "values": ["unknown", "green", "red"]
    }, {
        "id": "31",
        "name_fi": "Olen rollaattorin käyttäjä",
        "name_sv": "Jag är en rollatoranvändare",
        "name_en": "I am a rollator user",
        "values": ["unknown", "green", "red"]
    }, {
        "id": "32",
        "name_fi": "Olen rollaattorin käyttäjä - saavun omalla autolla",
        "name_sv": "Jag är en rollatoranvändare - kommer med min egen bil",
        "name_en": "I am a rollator user - arrive by my car",
        "values": ["unknown", "green", "red"]
    }, {
        "id": "33",
        "name_fi": "Olen rollaattorin käyttäjä - saavun saattoliikenteellä",
        "name_sv": "Jag är en rollatoranvändare - kommer med angöringstrafik",
        "name_en": "I am a rollator user - arrive with pick-up and drop-off traffic",
        "values": ["unknown", "green", "red"]
    }, {
        "id": "41",
        "name_fi": "Minulla on lastenvaunut",
        "name_sv": "Jag är en barnvagnsdragare",
        "name_en": "I am a stroller pusher",
        "values": ["unknown", "green", "red"]
    }, {
        "id": "51",
        "name_fi": "Olen näkövammainen",
        "name_sv": "Jag är synskadad",
        "name_en": "I am visually impaired",
        "values": ["unknown", "green", "red"]
    }, {
        "id": "52",
        "name_fi": "Olen näkövammainen - saavun saattoliikenteellä",
        "name_sv": "Jag är synskadad - kommer med angöringstrafik",
        "name_en": "I am visually impaired - arrive with pick-up and drop-off traffic",
        "values": ["unknown", "green", "red"]
    }, {
        "id": "61",
        "name_fi": "Käytän kuulolaitetta",
        "name_sv": "Jag använder en hörapparat",
        "name_en": "I use a hearing aid",
        "values": ["unknown", "green", "red"]
    }
]
```
</details>

#### TPREK Unit
https://www.hel.fi/palvelukarttaws/rest/v4/unit/5/?format=json&newfeatures=yes

<summary>Results</summary>
<details>

```json
{
    "id": 5,
    "name_fi": "Päiväkoti Kaunokki",
    "name_sv": "Päiväkoti Kaunokki",
    "name_en": "Daycare Kaunokki",
    "accessibility_viewpoints": "00:unknown,11:red,12:red,13:red,21:red,22:red,23:red,31:red,32:red,33:red,41:green,51:red,52:red,61:unknown"
}
```
</details>

#### TPREK Accessibility shortages
https://www.hel.fi/palvelukarttaws/rest/v4/unit/5/accessibility_shortage/

<summary>Results</summary>
<details>

```json
[{
        "viewpoint_id": "11",
        "shortage_fi": "Kulkureitti sisäänkäynnille ei ole tasainen.",
        "shortage_sv": "Rutten till ingången är inte jämn.",
        "shortage_en": "The route to the entrance is not even."
    }, {
        "viewpoint_id": "11",
        "shortage_fi": "Kulkureitillä sisäänkäynnille on jyrkkä mäki.",
        "shortage_sv": "På rutten till ingången finns en brant backe.",
        "shortage_en": "The route to the entrance has a steep slope."
    }, {
        "viewpoint_id": "11",
        "shortage_fi": "Sisäänkäynnissä on yli 2 cm korkea kynnys.",
        "shortage_sv": "Vid ingången finns en över 2 cm hög tröskel.",
        "shortage_en": "The entrance has a threshold over 2 cm high."
    }, {
        "viewpoint_id": "11",
        "shortage_fi": "Sisäänkäynnissä on ahdas tuulikaappi.",
        "shortage_sv": "Det finns ett trångt vindfång vid ingången.",
        "shortage_en": "The entrance has a cramped foyer."
    }, {
        "viewpoint_id": "12",
        "shortage_fi": "Esteettömiä autopaikkoja ei ole.",
        "shortage_sv": "Inga p-platser för personer med rörelsehinder.",
        "shortage_en": "No parking places for persons with a disability."
    }, {
        "viewpoint_id": "12",
        "shortage_fi": "Kulkureitti sisäänkäynnille ei ole tasainen.",
        "shortage_sv": "Rutten till ingången är inte jämn.",
        "shortage_en": "The route to the entrance is not even."
    }, {
        "viewpoint_id": "12",
        "shortage_fi": "Kulkureitillä sisäänkäynnille on jyrkkä mäki.",
        "shortage_sv": "På rutten till ingången finns en brant backe.",
        "shortage_en": "The route to the entrance has a steep slope."
    }, {
        "viewpoint_id": "12",
        "shortage_fi": "Sisäänkäynnissä on yli 2 cm korkea kynnys.",
        "shortage_sv": "Vid ingången finns en över 2 cm hög tröskel.",
        "shortage_en": "The entrance has a threshold over 2 cm high."
    }, {
        "viewpoint_id": "12",
        "shortage_fi": "Sisäänkäynnissä on ahdas tuulikaappi.",
        "shortage_sv": "Det finns ett trångt vindfång vid ingången.",
        "shortage_en": "The entrance has a cramped foyer."
    }, {
        "viewpoint_id": "13",
        "shortage_fi": "Kulkureitti sisäänkäynnille ei ole tasainen.",
        "shortage_sv": "Rutten till ingången är inte jämn.",
        "shortage_en": "The route to the entrance is not even."
    }, {
        "viewpoint_id": "13",
        "shortage_fi": "Kulkureitillä sisäänkäynnille on jyrkkä mäki.",
        "shortage_sv": "På rutten till ingången finns en brant backe.",
        "shortage_en": "The route to the entrance has a steep slope."
    }, {
        "viewpoint_id": "13",
        "shortage_fi": "Sisäänkäynnissä on yli 2 cm korkea kynnys.",
        "shortage_sv": "Vid ingången finns en över 2 cm hög tröskel.",
        "shortage_en": "The entrance has a threshold over 2 cm high."
    }, {
        "viewpoint_id": "13",
        "shortage_fi": "Sisäänkäynnissä on ahdas tuulikaappi.",
        "shortage_sv": "Det finns ett trångt vindfång vid ingången.",
        "shortage_en": "The entrance has a cramped foyer."
    }, {
        "viewpoint_id": "21",
        "shortage_fi": "Kulkureitti sisäänkäynnille ei ole tasainen.",
        "shortage_sv": "Rutten till ingången är inte jämn.",
        "shortage_en": "The route to the entrance is not even."
    }, {
        "viewpoint_id": "21",
        "shortage_fi": "Kulkureitillä sisäänkäynnille on jyrkkä mäki.",
        "shortage_sv": "På rutten till ingången finns en brant backe.",
        "shortage_en": "The route to the entrance has a steep slope."
    }, {
        "viewpoint_id": "22",
        "shortage_fi": "Esteettömiä autopaikkoja ei ole.",
        "shortage_sv": "Inga p-platser för personer med rörelsehinder.",
        "shortage_en": "No parking places for persons with a disability."
    }, {
        "viewpoint_id": "22",
        "shortage_fi": "Kulkureitti sisäänkäynnille ei ole tasainen.",
        "shortage_sv": "Rutten till ingången är inte jämn.",
        "shortage_en": "The route to the entrance is not even."
    }, {
        "viewpoint_id": "22",
        "shortage_fi": "Kulkureitillä sisäänkäynnille on jyrkkä mäki.",
        "shortage_sv": "På rutten till ingången finns en brant backe.",
        "shortage_en": "The route to the entrance has a steep slope."
    }, {
        "viewpoint_id": "23",
        "shortage_fi": "Kulkureitti sisäänkäynnille ei ole tasainen.",
        "shortage_sv": "Rutten till ingången är inte jämn.",
        "shortage_en": "The route to the entrance is not even."
    }, {
        "viewpoint_id": "23",
        "shortage_fi": "Kulkureitillä sisäänkäynnille on jyrkkä mäki.",
        "shortage_sv": "På rutten till ingången finns en brant backe.",
        "shortage_en": "The route to the entrance has a steep slope."
    }, {
        "viewpoint_id": "31",
        "shortage_fi": "Kulkureitti sisäänkäynnille ei ole tasainen.",
        "shortage_sv": "Rutten till ingången är inte jämn.",
        "shortage_en": "The route to the entrance is not even."
    }, {
        "viewpoint_id": "31",
        "shortage_fi": "Kulkureitillä sisäänkäynnille on jyrkkä mäki.",
        "shortage_sv": "På rutten till ingången finns en brant backe.",
        "shortage_en": "The route to the entrance has a steep slope."
    }, {
        "viewpoint_id": "31",
        "shortage_fi": "Sisäänkäynnissä on yli 2 cm korkea kynnys.",
        "shortage_sv": "Vid ingången finns en över 2 cm hög tröskel.",
        "shortage_en": "The entrance has a threshold over 2 cm high."
    }, {
        "viewpoint_id": "31",
        "shortage_fi": "Sisäänkäynnissä on ahdas tuulikaappi.",
        "shortage_sv": "Det finns ett trångt vindfång vid ingången.",
        "shortage_en": "The entrance has a cramped foyer."
    }, {
        "viewpoint_id": "32",
        "shortage_fi": "Esteettömiä autopaikkoja ei ole.",
        "shortage_sv": "Inga p-platser för personer med rörelsehinder.",
        "shortage_en": "No parking places for persons with a disability."
    }, {
        "viewpoint_id": "32",
        "shortage_fi": "Kulkureitti sisäänkäynnille ei ole tasainen.",
        "shortage_sv": "Rutten till ingången är inte jämn.",
        "shortage_en": "The route to the entrance is not even."
    }, {
        "viewpoint_id": "32",
        "shortage_fi": "Kulkureitillä sisäänkäynnille on jyrkkä mäki.",
        "shortage_sv": "På rutten till ingången finns en brant backe.",
        "shortage_en": "The route to the entrance has a steep slope."
    }, {
        "viewpoint_id": "32",
        "shortage_fi": "Sisäänkäynnissä on yli 2 cm korkea kynnys.",
        "shortage_sv": "Vid ingången finns en över 2 cm hög tröskel.",
        "shortage_en": "The entrance has a threshold over 2 cm high."
    }, {
        "viewpoint_id": "32",
        "shortage_fi": "Sisäänkäynnissä on ahdas tuulikaappi.",
        "shortage_sv": "Det finns ett trångt vindfång vid ingången.",
        "shortage_en": "The entrance has a cramped foyer."
    }, {
        "viewpoint_id": "33",
        "shortage_fi": "Kulkureitti sisäänkäynnille ei ole tasainen.",
        "shortage_sv": "Rutten till ingången är inte jämn.",
        "shortage_en": "The route to the entrance is not even."
    }, {
        "viewpoint_id": "33",
        "shortage_fi": "Kulkureitillä sisäänkäynnille on jyrkkä mäki.",
        "shortage_sv": "På rutten till ingången finns en brant backe.",
        "shortage_en": "The route to the entrance has a steep slope."
    }, {
        "viewpoint_id": "33",
        "shortage_fi": "Sisäänkäynnissä on yli 2 cm korkea kynnys.",
        "shortage_sv": "Vid ingången finns en över 2 cm hög tröskel.",
        "shortage_en": "The entrance has a threshold over 2 cm high."
    }, {
        "viewpoint_id": "33",
        "shortage_fi": "Sisäänkäynnissä on ahdas tuulikaappi.",
        "shortage_sv": "Det finns ett trångt vindfång vid ingången.",
        "shortage_en": "The entrance has a cramped foyer."
    }, {
        "viewpoint_id": "41",
        "shortage_fi": "Ei puutteita.",
        "shortage_sv": "Inga brister.",
        "shortage_en": "No shortcomings."
    }, {
        "viewpoint_id": "51",
        "shortage_fi": "Sisätilojen kulkureiteillä ei ole liikkumista ohjaavaa pintamateriaalia (tummuus- tai tuntokontrasti).",
        "shortage_sv": "Rutterna inomhus saknar ytmaterial ger anvisningar om hur man tar sig fram (ljushets- eller materialkontrast).",
        "shortage_en": "The indoor routes do not have guiding surface materials (colour or tactile contrast)."
    }, {
        "viewpoint_id": "51",
        "shortage_fi": "Sisätilojen lasiovissa ei ole kontrastimerkintöjä.",
        "shortage_sv": "Det finns inga kontrastmarkeringar på glasdörrarna inomhus.",
        "shortage_en": "The glass doors in the indoor area do not have contrast markings."
    }, {
        "viewpoint_id": "52",
        "shortage_fi": "Sisätilojen kulkureiteillä ei ole liikkumista ohjaavaa pintamateriaalia (tummuus- tai tuntokontrasti).",
        "shortage_sv": "Rutterna inomhus saknar ytmaterial ger anvisningar om hur man tar sig fram (ljushets- eller materialkontrast).",
        "shortage_en": "The indoor routes do not have guiding surface materials (colour or tactile contrast)."
    }, {
        "viewpoint_id": "52",
        "shortage_fi": "Sisätilojen lasiovissa ei ole kontrastimerkintöjä.",
        "shortage_sv": "Det finns inga kontrastmarkeringar på glasdörrarna inomhus.",
        "shortage_en": "The glass doors in the indoor area do not have contrast markings."
    }, {
        "viewpoint_id": "61",
        "shortage_fi": "Ei tiedettyjä puutteita.",
        "shortage_sv": "Inga kända brister.",
        "shortage_en": "No known shortcomings."
    }
]
```
</details>

#### Servicemap Unit
https://api.hel.fi/servicemap/v2/unit/5/

<summary>Results</summary>
<details>

```json
{
    "id": 5,
    "name": {
        "fi": "Päiväkoti Kaunokki",
        "sv": "Päiväkoti Kaunokki",
        "en": "Daycare Kaunokki"
    },
    "accessibility_viewpoints": {
        "00": null,
        "11": "red",
        "12": "red",
        "13": "red",
        "21": "red",
        "22": "red",
        "23": "red",
        "31": "red",
        "32": "red",
        "33": "red",
        "41": "green",
        "51": "red",
        "52": "red",
        "61": null
    },
    "accessibility_shortcoming_count": {
        "rollator": 5,
        "wheelchair": 5,
        "reduced_mobility": 3,
        "visually_impaired": 2
    }
}
```
</details>

#### So what got changed in the data?

Accessibility shortcoming counts:
- **"hearing_aid": null**
  - i.e. explicitly unknown now in output data, wasn't present before in accessibility_shortcoming_count or accessibility.shortcomings
- "rollator": 5
- **"stroller": 0**
  - i.e. explicitly no shortcomings now in output data, wasn't present before in accessibility_shortcoming_count or accessibility.shortcomings
- "wheelchair": 5
- "reduced_mobility": 3
- "visually_impaired": 2

[US-112]: https://helsinkisolutionoffice.atlassian.net/browse/US-112?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by terovirtanen** [comment]
**Unified search graphql branch is deployed to platta: https://kuva-unified-search-pr107.api.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Unified search sources branch is deployed to platta: https://kuva-unified-search-sources-pr107.api.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Unified search graphql branch is deployed to platta: https://kuva-unified-search-pr107.api.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Unified search sources branch is deployed to platta: https://kuva-unified-search-sources-pr107.api.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Unified search graphql branch is deployed to platta: https://kuva-unified-search-pr107.api.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Unified search sources branch is deployed to platta: https://kuva-unified-search-sources-pr107.api.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Unified search graphql branch is deployed to platta: https://kuva-unified-search-pr107.api.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Unified search sources branch is deployed to platta: https://kuva-unified-search-sources-pr107.api.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Unified search graphql branch is deployed to platta: https://kuva-unified-search-pr107.api.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Unified search sources branch is deployed to platta: https://kuva-unified-search-sources-pr107.api.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Unified search graphql branch is deployed to platta: https://kuva-unified-search-pr107.api.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Unified search graphql branch is deployed to platta: https://kuva-unified-search-pr107.api.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Unified search graphql branch is deployed to platta: https://kuva-unified-search-pr107.api.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Unified search sources branch is deployed to platta: https://kuva-unified-search-sources-pr107.api.dev.hel.ninja** :rocket::rocket::rocket:

File: sources/ingest/importers/location.py [hunk 1]
```diff
@@ -197,6 +276,77 @@ def set_accessibility_shortages(
         for viewpoint in self.viewpoints:
             viewpoint.shortages = viewpoint_id_to_shortages.get(viewpoint.id, [])[:]
 
+    def fix_unknown_and_zero_shortcomings(self) -> bool:
```
Comment (nikomakela) [inline sources/ingest/importers/location.py:279]: I would split this (rather long) function to multiple private functions just to make it more clean. 🍝

**Review by nikomakela — APPROVED** [review]
The long function should be splitted to multiple parts. Rest is golden. Good tests!

File: sources/ingest/importers/location.py (hunk 1 above)
Comment (karisal-anders) [reply sources/ingest/importers/location.py:279]: Taking into account in another PR with refactoring

**Pull request merged**